command/src/config.rs

```diff
@@ -1465,24 +1465,33 @@ impl Config {
         };
 
         debug!("saved_stated path in the config: {}", path);
-
         let config_path_buf = PathBuf::from(self.config_path.clone());
-        debug!("Config path buffer: {:?}", config_path_buf);
 
+        debug!("Config path buffer: {:?}", config_path_buf);
         let config_folder = config_path_buf
             .parent()
             .with_context(|| "could not get parent folder of configuration file")?;
 
         debug!("Config folder: {:?}", config_folder);
+        if !config_folder.exists() {
+            create_dir_all(config_folder).with_context(|| format!("failed to create state parent directory '{}'", config_folder.display()))?;
+        }
```

Comment:
the config folder is computed from the full path of the config file, see line 1472.. It *must* exist already, or the config file wouldn't exist at all.

also, please apply cargo fmt

Reply:
In few case, the absolute path one, we could compute the path from the string, but it could not exists on filesystem. For example, in the case of packaging (see #945), we could compute the path if the state is located at `/var/lib/sozu/state.json`, but the folder `/var/lib/sozu` does not exists. So, we have to create it.